functorch/_src/aot_autograd.py

```diff
@@ -1457,7 +1499,36 @@ def compiled_function(*args):
         else:
             return fw_outs
 
-    return compiled_function
+    if not config.debug_assert:
+        return compiled_function
+
+    flat_requires_grad = [a.requires_grad if isinstance(a, Tensor) else None for a in flat_args]
+
+    @wraps(compiled_function)
+    def debug_compiled_function(*args):
+        # TODO: Check aliasing relationships
+        # TODO: Check strides for metadata mutation
```

Comment:
Checking my understanding - is this saying that we should check that the strides of inputs are exactly the same across runs? That seems at odds with our goal of eventually being stride agnostic.

Reply:
What I mean here, is if we make assumptions about what the input strides of the tensor should be, we should test those assumptions are upheld. This may not be an exact test; in dynamic shapes, this would correspond to testing that the symbolic formulas match up, but in static shapes it would be an exact test. In any case, the point is to avoid accuracy errors from unchecked `as_strided` invocation